src/controllers/translation.ts

```diff
@@ -32,43 +37,48 @@ const IGBO_ENGLISH_TRANSLATION_INPUT_MAX_LENGTH = 120;
  */
 export const getTranslation: MiddleWare = async (req, res, next) => {
   try {
-    console.log('entered translation endpoint');
-    console.log('request body', req.body);
     const requestBodyValidation = TranslationRequestBody.safeParse(req.body);
     if (!requestBodyValidation.success) {
       throw fromError(requestBodyValidation.error);
     }
 
     const requestBody = requestBodyValidation.data;
-    console.log(`validation succeeded, body: ${requestBody}`);
 
     if (requestBody.sourceLanguageCode === requestBody.destinationLanguageCode) {
       throw new Error('Source and destination languages must be different');
     }
+
     if (
-      requestBody.sourceLanguageCode !== LanguageEnum.IGBO ||
-      requestBody.destinationLanguageCode !== LanguageEnum.ENGLISH
+      !(
+        `${requestBody.sourceLanguageCode}to${requestBody.destinationLanguageCode}` in
+        SUPPORTED_TRANSLATIONS
+      )
     ) {
       throw new Error(
         `${requestBody.sourceLanguageCode} to ${requestBody.destinationLanguageCode} translation is not yet supported`
       );
     }
-    const igboText = requestBody.text;
-    if (!igboText) {
+    const textToTranslate = requestBody.text;
+    if (!textToTranslate) {
       throw new Error('Cannot translate empty string');
     }
-
-    if (igboText.length > IGBO_ENGLISH_TRANSLATION_INPUT_MAX_LENGTH) {
-      throw new Error('Cannot translate text greater than 120 characters');
+    const translationKey =
```

Comment:
Instead of casting these strings together to create a key to determine the payload shape, I think the `sourceLanguageCode` could map to the desired payload shape. Something like

```ts
const PayloadKeyMap = {
  [LanguageEnum.Igbo]: 'igbo',
  [LanguageEnum.English]: 'english'
}
```

That way, we could take the value of indexing into that map to determine the final payload shape:

```ts
const payload = { [PayloadKeyMap[requestBody.sourceLanguageCode]: textToTranslate }
```

If `requestBody.sourceLanguageCode === 'ibo'` then we would get a final payload of `{ igbo: textToTranslate }` (if my logic is correct 😅 )

I think there's an even cleaner way of us handling this, but I think relying on maps rather than strings to determine payload shapes is a more stable way to go

Reply:
Switched to something like this. Thanks for the suggestion!